bioimageio/core/workflow/operators/_run.py

```diff
@@ -0,0 +1,202 @@
+from dataclasses import dataclass
+from os import PathLike
+from typing import Any, Dict, Generator, IO, List, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import xarray as xr
+from marshmallow import missing
+
+from bioimageio.core import load_resource_description
+from bioimageio.core.prediction_pipeline import create_prediction_pipeline
+from bioimageio.core.resource_io import nodes
+from bioimageio.spec import load_raw_resource_description
+from bioimageio.spec.model import raw_nodes
+from bioimageio.spec.shared.raw_nodes import ResourceDescription as RawResourceDescription
+
+
+def run_model_inference(
+    rdf_source: Union[dict, PathLike, IO, str, bytes, raw_nodes.URI, RawResourceDescription],
+    *tensors,
+    enable_preprocessing: bool = True,
+    enable_postprocessing: bool = True,
+    devices: Optional[Sequence[str]] = None,
+) -> List[xr.DataArray]:
+    """run model inference
+
+    Returns:
+        list: model outputs
+    """
+    model = load_raw_resource_description(rdf_source, update_to_format="latest")
+    assert isinstance(model, raw_nodes.Model)
+    # remove pre-/postprocessing if specified
+    if not enable_preprocessing:
+        for ipt in model.inputs:
+            if ipt.preprocessing:
+                ipt.preprocessing = missing
+    if not enable_postprocessing:
+        for ipt in model.outputs:
+            if ipt.postprocessing:
+                ipt.postprocessing = missing
+
+    with create_prediction_pipeline(model, devices=devices) as pred_pipeline:
+        return pred_pipeline.forward(*tensors)
+
+
+def run_workflow(
+    rdf_source: Union[dict, PathLike, IO, str, bytes, raw_nodes.URI, RawResourceDescription],
+    inputs: Sequence = tuple(),
+    options: Dict[str, Any] = None,
+) -> tuple:
+    outputs = tuple()
+    for state in _iterate_workflow_steps_impl(rdf_source, test_steps=False, inputs=inputs, options=options):
+        outputs = state.outputs
+
+    return outputs
+
+
+def run_workflow_test(
+    rdf_source: Union[dict, PathLike, IO, str, bytes, raw_nodes.URI, RawResourceDescription],
+) -> tuple:
+    outputs = tuple()
+    for state in _iterate_workflow_steps_impl(rdf_source, test_steps=True):
+        outputs = state.outputs
+
+    return outputs
+
+
+@dataclass
+class WorkflowState:
+    wf_inputs: Dict[str, Any]
+    wf_options: Dict[str, Any]
+    inputs: tuple
+    outputs: tuple
+    named_outputs: Dict[str, Any]
+
+
+def iterate_workflow_steps(
+    rdf_source: Union[dict, PathLike, IO, str, bytes, raw_nodes.URI, RawResourceDescription],
+    *,
+    inputs: Sequence = tuple(),
+    options: Dict[str, Any] = None,
+) -> Generator[WorkflowState]:
+    yield from _iterate_workflow_steps_impl(rdf_source, inputs=inputs, options=options, test_steps=False)
```

Comment:
@oeway would this separate op suffice?
If the behavior only depends on a flag the return type of `run_workflow` changes somewhat drastically from an output tuple to a generator (of intermediate output tuples or--like I implemented now--a state object).